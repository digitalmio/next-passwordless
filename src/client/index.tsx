// import { useState, useEffect } from 'react';

const fetchSession = async () => {
  const data = await fetch('/api/auth/session').then(res => res.json());

  return {
    success: Boolean(data),
    user: data,
  };
};
