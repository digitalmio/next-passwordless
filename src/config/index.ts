export interface IConfig {
  secret: string;
  rootUrl: string;
  linkExpiry?: string;
  showLinkOnDev?: boolean;
  sendEmailsOnDev?: boolean;
}

export const defaultConfig: Required<IConfig> = {
  secret: '',
  rootUrl: '',
  linkExpiry: '1h',
  showLinkOnDev: true,
  sendEmailsOnDev: false,
};
