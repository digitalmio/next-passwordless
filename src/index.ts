import { NextApiRequest, NextApiResponse } from 'next';
import { processCallback } from './actions/processCallback';
import { sendToken } from './actions/sendToken';
import { parseConfig, IConfig } from './config';

export const NextPasswordless = (userConfig: IConfig) => (
  req: NextApiRequest,
  res: NextApiResponse
): Promise<void> | void => {
  const { nextPasswordless: path } = req.query;
  const { method } = req;

  // merge user config with defaults
  const config = parseConfig(userConfig);

  // check for empty secret
  if (!config.secret) {
    throw new Error('You need to provide JWT secret.');
  }

  // root path (so path is undefined), post to generate and send token
  if (!path && method === 'POST') return sendToken(config, req, res);

  // callback url
  if (path?.length === 1 && path[0] === 'callback' && method === 'GET')
    return processCallback(config, req, res);

  // defaults to 404
  return res.status(404).json({ status: 404, message: 'Page not found' });
};
