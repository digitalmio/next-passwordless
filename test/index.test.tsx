import { NextPasswordless } from '../src';

describe('root', () => {
  it('should snapshot main function', () => {
    expect(NextPasswordless({ secret: 'abc123' })).toMatchSnapshot();
  });
});
