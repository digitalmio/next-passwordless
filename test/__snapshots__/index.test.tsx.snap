// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root should snapshot main function 1`] = `[Function]`;
